### ISSUE TITLE
1.3.1 changed the hash，not sync with html-webpack-plugin

### ISSUE THREAD

**Issue by wulucxy** [scinos/webpack-plugin-hash-output#24]
@scinos 
the problem is here，when I update 1.3.1，my html hash was totally changed。

```javascript
plugins: [
  new HtmlWebpackPlugin({
      filename: 'index.html',
      template: 'public/index.html',
      chunks: ['manifest', 'vendor', 'app']
    }),
   new WebpackPluginHashOutput()
]
```
then the output html hash has been changed。the reference script url is not right。

but as [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) said:

> If you have plugins that make use of it, html-webpack-plugin should be ordered first before any of the integrated plugins.

I an confused here how to set the order here

**Comment by scinos** [comment]
I'll take a look ASAP

**Comment by sunkeysun** [comment]
I also encounter this issue on version 3.1.1, but version 3.1.0 not has this problem.

**Comment by scinos** [comment]
@wulucxy, @sunkeysun  could you try with 

```
plugins: [
  new WebpackPluginHashOutput(),
  new HtmlWebpackPlugin({
      filename: 'index.html',
      template: 'public/index.html',
      chunks: ['manifest', 'vendor', 'app']
    })
]
```
(i.e. webpack-plugin-hash-output _before_ html-webpack-plugin)?

According to my test this should fix the problem. If that's the case, I'll update the README and, if possible, try to detect the wrong order and print a warning.

**Comment by wulucxy** [comment]
@scinos this is OK。but how is this happen?

**Comment by scinos** [comment]
TLDR: we changed the webpack phase where the plugin runs and now it competes with `html-webpack-plugin`

So in v 3.1.1 and 1.3.1 we changed the webpack hook used to run the plugin from `after-optimize-chunk-assets` to `emit`. The reason for using `after-optimize-chunk-assets` in the past was mainly because I didn't know better (😩), I should have used `emit` form the beginning because it makes more sense. This was also causing bugs with other plugins (#22).

So now it runs in `emit`. But there are other plugins that _also_ run on `emit`, like `html-webpack-plugin`). I do have a test for `html-webpack-plugin`, and because it run fine, I didn't consider this could break anything, my bad.

For the plugin to work it has to be the first plugin in the `emit` phase (which is determined by the order of the plugins in your `webpack.config.js`.

**Comment by scinos** [comment]
I released a new version (3.2.0) that will print a warning if there are other plugins that might interfere with us:

```
[WARNING] There are other plugins configured that might interfere with
webpack-plugin-hash-output. For this plugin to work correctly, it should
be the first plugin in the "emit" phase. Please be sure that the rest of the
plugins run after webpack-plugin-hash-output (ensure they are listed after
webpack-plugin-hash-output in the 'plugins' option in your webpack config).

Plugins that could interfere with webpack-plugin-hash-output:
 * HtmlWebpackPlugin
```
